### PR TITLE
chore(deps): Upgrade to Django 5

### PR DESCRIPTION
**Pull request by webjunkie**
## Problem

Let's see how much trouble it is.

Update: Seems it's not much trouble 🎉 

## Changes

- upgrade
- fix some timezone issue
- fix some types
- fix logout - GET is deprecated

## Does this work well for both Cloud and self-hosted?

n/a

## How did you test this code?

- relying on all unit and e2e tests here

### CODE REVIEW

File: posthog/middleware.py [hunk 1]
```diff
@@ -608,6 +608,8 @@ def process_response(self, request, response):
             # clears the cookies that were previously set, except for ph_current_instance as that is used for the website login button
             response.delete_cookie("ph_current_project_token", domain=default_cookie_options["domain"])
             response.delete_cookie("ph_current_project_name", domain=default_cookie_options["domain"])
+            return response
```
Comment (webjunkie) [inline posthog/middleware.py:611]: It's interesting that this is suddenly necessary... and it seems more correct?
@benjackwhite maybe you can have a look?
also see the needed proper logout in `posthog/middleware.py`

Comment (webjunkie) [reply posthog/middleware.py:612]: Ping 😊

**Review by benjackwhite — APPROVED** [review]
Generally looks good. The datetime changes are a bit scary as I wonder if there are any other weird places it was missed. We likely will want to do a very robust canary test with this...

File: cypress/productAnalytics/index.ts [hunk 2]
```diff
@@ -1,5 +1,13 @@
 import { urls } from 'scenes/urls'
 
+export const auth = {
```
[the remark this answers is not in the document]
Comment (webjunkie) [reply cypress/productAnalytics/index.ts:3]: Oops, true.

**Pull request merged**